### PR TITLE
feat: add `purgeCache` helper

### PR DESCRIPTION
**Pull request by eduardoboucas**
**Which problem is this pull request solving?**

Adds a `purgeCache` helper for https://www.notion.so/netlify/Cache-Purge-API-12b8eb7359c549a4aad56d528f19feb0, using the environment variable added in https://github.com/netlify/serverless-functions-api/pull/161.

I still need to add some tests, but wanted to get the PR up sooner rather than later to get feedback on the approach.

### CODE REVIEW

File: src/lib/purge_cache.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  const siteID = env.SITE_ID || overrides.siteID
+
+  if (!siteID) {
+    throw new Error(
+      'The Netlify site ID was not found in the execution environment. Please supply it manually using the `siteID` property.',
+    )
+  }
+
+  const token = env.NETLIFY_PURGE_TOKEN_1 || overrides.token
```
Comment (paulo) [inline src/lib/purge_cache.ts:25]: It's `NETLIFY_PURGE_API_TOKEN` now, right?

File: src/lib/purge_cache.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  if (!siteID) {
```
Comment (paulo) [inline src/lib/purge_cache.ts:27]: !siteID -> !token

File: src/lib/purge_cache.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { env } from 'process'
+
+interface PurgeCacheOptions {
+  apiURL?: string
+  siteID?: string
+  tags: string[]
```
Comment (paulo) [inline src/lib/purge_cache.ts:6]: The API can take all of these arguments (except `paths`): https://www.notion.so/netlify/Cache-Purge-API-12b8eb7359c549a4aad56d528f19feb0?pvs=4#46c26a47bd3a4689bb38dcd97c7b6eae

File: src/lib/purge_cache.ts (hunk 2 above)
Comment (eduardoboucas) [reply src/lib/purge_cache.ts]: Oops! Fixed.

File: src/lib/purge_cache.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    cache_tags: options.tags,
+    site_id: env.SITE_ID,
+  }
+  const token = env.NETLIFY_PURGE_API_TOKEN || options.token
+
+  if ('siteSlug' in options) {
+    payload.deploy_alias = options.deployAlias
+    payload.site_slug = options.siteSlug
+  } else if ('domain' in options) {
+    payload.domain = options.domain
```
Comment (paulo) [inline src/lib/purge_cache.ts:50]: Why can't we provide the `deployAlias` for this option?

**Review by paulo — DISMISSED** [review]

**Review by paulo — APPROVED** [review]

**Review by minivan — APPROVED** [review]

**Pull request merged**